### DISCUSSION TITLE
Fundamentos do Blazor Server - Ciclo de vida com render mode

### DISCUSSION THREAD

**Discussion by lucas-softata** [balta-io/forum#531] [Tira Dúvidas] [answered]
Tenho percebido que ao iniciar o clico de vida utilizando @rendermode InteractiveServer o método OnInitialized() é executado duas vezes, sendo assim, o consumo que faço em meu repositório de dados é executado duas vezes de acordo com as informações que aprece no log do debug, é normal? é executado realmente duas vezes / uma vez ou tem algo a ser feito para corrigir/sanar este problema? Pois preciso carregar os dados no momento em que inicializa o componente/página.

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @lucas-softata, como vai?

Você seguiu os mesmos passos do curso? Pergunto pois existem vários sabores e formas de renderizar conteúdo no Blazor, que podem afetar este comportamento.

Se estiver com o PreRender ligado e executar o InteractiveServer, ele vai passar duas vezes... mas a explicação depende de como inicializou o projeto e como está chamando seu componente ou página com o rendermode em questão.

Me detalha mais que consigo te ajudar melhor...
